### PR TITLE
Segregate signatures from transaction bodies

### LINKED ISSUE #510: Transaction hash should not depend on the signatures field
A.k.a implement segwit.

A naive solution would be to set the signatures field to an empty vector before hashing:

```rust
impl Hashable for Transaction {
    fn hash(&self) -> Hash {
        let th = Transaction { signatures: vec![], ..self.clone() };
        calculate_sha256(&th.to_pb_bytes().unwrap()).into()
    }
}
```

This is a breaking change.

### PR DESCRIPTION
**Pull request by aesedepece**
Also implements hash caching for `Transaction` / `TransactionBody` (hashes only get computed once).

Fix #510

### CODE REVIEW

File: node/src/actors/messages.rs [hunk 1]
```diff
@@ -13,7 +13,7 @@ use std::{
 use witnet_data_structures::{
     chain::{
         Block, CheckpointBeacon, Epoch, Hash, InventoryEntry, InventoryItem, RADConsensus,
-        RADRequest, Transaction,
```
Comment (lrubiorod) [inline node/src/actors/messages.rs]: You remove here and import in line 35

Comment (aesedepece) [reply node/src/actors/messages.rs]: F*ck CLion.

File: node/src/actors/session/handlers.rs [hunk 2]
```diff
@@ -11,7 +11,7 @@ use log::{debug, error, info, trace, warn};
 
 use witnet_data_structures::{
     builders::from_address,
-    chain::{Block, CheckpointBeacon, Hashable, InventoryEntry, InventoryItem, Transaction},
```
Comment (lrubiorod) [inline node/src/actors/session/handlers.rs]: Here again xD

File: validations/src/validations.rs [hunk 3]
```diff
@@ -4,21 +4,22 @@ use std::collections::HashMap;
 use witnet_data_structures::{
     chain::{
         Block, BlockError, BlockInChain, CheckpointBeacon, DataRequestOutput, Epoch, Hash,
-        Hashable, Input, Output, OutputPointer, RADRequest, Transaction, TransactionError,
```
Comment (lrubiorod) [inline validations/src/validations.rs]: One more

**Pull request merged**